stdlib/public/Concurrency/Actor.cpp

```diff
@@ -1642,9 +1647,6 @@ void ProcessOutOfLineJob::process(Job *job) {
   DefaultActorImpl *actor = self->Actor;
 
   delete self;
-
-  // Balances with the swift_release in defaultActorDrain()
-  swift_retain(actor);
```

Comment:
It is ok for actor to be deallocated while job is running on its executor, logic around `Zombie_ReadyForDeallocation` state takes care of this. But looks like this scenario was never covered by the unit tests. With isolated deinit is became possible to reproduce this scenario reliably.

One of the unit tests that I've added covers this, and detected a bug - actor was deallocated with +1 retain count triggering `fatalError()`. If control could reach `swift_release()`, it would be called on deallocated pointer.

Reply:
Sounds good, thanks for the explanation